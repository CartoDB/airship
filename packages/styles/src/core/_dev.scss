@import 'tools/_mixins';
@import 'variables/_variables';
