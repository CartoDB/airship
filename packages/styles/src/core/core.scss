@import 'variables/_variables';

* {
  box-sizing: border-box;
  font-family: sans-serif;
}

:root {
  --primary-color: #{$primary-color};
}
