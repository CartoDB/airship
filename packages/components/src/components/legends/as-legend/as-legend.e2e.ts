import { E2EPage, newE2EPage } from '@stencil/core/testing';

describe('as-legend', async () => {
  describe('Rendering', async () => {
    let page: E2EPage;

    it('should render a heading and description', async () => {
      page = await newE2EPage({ html: `
        <as-legend heading="Heading" description="Description">
        </as-legend>
      ` });

      const element = await page.find('as-legend');
      expect(element.outerHTML).toMatchSnapshot();
    });

    it('should not render the heading element if no heading and description are provided', async () => {
      page = await newE2EPage({ html: `
        <as-legend>
        </as-legend>
      ` });

      const element = await page.find('as-widget-header');
      expect(element).toBeNull();
    });

    it('should not render content if loading', async () => {
      page = await newE2EPage({ html: `
        <as-legend loading>
        </as-legend>
      ` });

      const element = await page.find('as-legend');
      expect(element.outerHTML).toMatchSnapshot();
    });

    it('should support a slot for legends', async () => {
      page = await newE2EPage({ html: `
        <as-legend>
          <span slot="legends">Legend</span>
        </as-legend>
      ` });

      const element = await page.find('as-legend');
      expect(element.outerHTML).toMatchSnapshot();
    });

    it('should support a slot for the footer', async () => {
      page = await newE2EPage({ html: `
        <as-legend>
          <span slot="footer">Footer</span>
        </as-legend>
      ` });

      const element = await page.find('as-legend');
      expect(element.outerHTML).toMatchSnapshot();
    });
  });
});
