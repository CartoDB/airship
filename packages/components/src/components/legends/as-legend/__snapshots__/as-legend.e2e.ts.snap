// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`as-legend Rendering should not render content if loading 1`] = `"<as-legend loading=\\"\\" class=\\"hydrated\\"><div class=\\"as-legend--wrapper\\"></div></as-legend>"`;

exports[`as-legend Rendering should render a heading and description 1`] = `"<as-legend heading=\\"Heading\\" description=\\"Description\\" class=\\"hydrated\\"><div class=\\"as-legend--wrapper\\"><as-widget-header class=\\"hydrated\\"><h2 class=\\"as-widget-header__header\\">Heading</h2><p class=\\"as-widget-header__subheader as-body\\">Description</p></as-widget-header><div class=\\"as-legend--legends-slot\\"></div></div></as-legend>"`;

exports[`as-legend Rendering should support a slot for legends 1`] = `"<as-legend class=\\"hydrated\\"><div class=\\"as-legend--wrapper\\"><div class=\\"as-legend--legends-slot\\"><span slot=\\"legends\\">Legend</span></div></div></as-legend>"`;

exports[`as-legend Rendering should support a slot for the footer 1`] = `"<as-legend class=\\"hydrated\\"><div class=\\"as-legend--wrapper\\"><div class=\\"as-legend--legends-slot\\"></div><span slot=\\"footer\\">Footer</span></div></as-legend>"`;
